buku

```diff
@@ -4293,15 +4304,14 @@ def gen_headers():
     MYPROXY = os.environ.get('https_proxy')
     if MYPROXY:
         try:
-            url = parse_url(MYPROXY)
+            auth, MYPROXY = extract_auth(MYPROXY)
```

Comment:
The value returned by `urlparse()` doesn't have a joined `.auth` field (splitting it into `.username` and `.password`)